### PR TITLE
Fix add resource and refactor TreeView components

### PR DESCRIPTION
**Pull request by shimoncohen**
Fixed:
* adding resource to the map
* now using url as unique value instead of resource name
* moved some dependencies to dev

Refactor:
* tree view components are now generic

Added:
* resource level view
* styling configurations for prettier and commitlint

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "trailingComma": "es5",
+    "tabWidth": 2,
+    "semi": true,
+    "singleQuote": true
+  }
```
Comment (asafMasa) [inline .prettierrc]: enter

File: src/Components/Group/Groups.js [hunk 2]
```diff
@@ -15,6 +18,11 @@ export default React.memo(function Groups(props) {
     setData(dataFromStore);
   }, [dataFromStore]);
 
+  const fetchResource = async function (path, name) {
+    const res = await GetResource(path, name);
```
Comment (asafMasa) [inline src/Components/Group/Groups.js]: camelCase

File: src/Components/Group/Item/Item.js [hunk 3]
```diff
@@ -120,10 +125,10 @@ export default React.memo(function Item(props) {
             </ExpansionPanelDetails>
           </ExpansionPanel>
           <SettingsTabs
-            itemId={item.name}
+            // itemId={item.name}
```
Comment (asafMasa) [inline src/Components/Group/Item/Item.js]: delete comments

File: src/Components/Group/Item/OpacitySlider.js [hunk 4]
```diff
@@ -72,7 +72,7 @@ export default function CustomizedSlider(props) {
       setOpacity(newValue);
```
Comment (asafMasa) [inline src/Components/Group/Item/OpacitySlider.js:72]: rename CustomizedSlider to OpacitySlider

File: src/App.js [hunk 5]
```diff
@@ -1,15 +1,10 @@
 import React, { useEffect, useState } from "react";
 import "./App.css";
 import Map from "./Components/Map/Map";
-import Axios from "axios";
-import { useDispatch, useSelector } from "react-redux";
-import { useSnackbar } from "notistack";
-import Config from "./Config/urls";
+import { useSelector } from "react-redux";
```
Comment (asafMasa) [inline src/App.js]: '

File: src/Components/Group/Selector/TreeView.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    } catch {
+      enqueueSnackbar(ERROR_CONTACTING_SERVER.message, {
+        variant: ERROR_CONTACTING_SERVER.variant,
+      });
+    }
+  };
+
+  const renderItems = function (path, items) {
+    return items.map((item) => (
+      // <TreeItem
```
Comment (asafMasa) [inline src/Components/Group/Selector/TreeView.js]: remove remark

File: src/Components/Group/Selector/VersionSelect.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      const res = await itemRequest(path, name, version);
+      setItem(res);
+    } catch {
+      enqueueSnackbar(ERROR_CONTACTING_SERVER.message, {
+        variant: ERROR_CONTACTING_SERVER.variant,
+      });
+    }
+  };
+
+  //   const selectItem = async function (version) {
```
Comment (asafMasa) [inline src/Components/Group/Selector/VersionSelect.js]: remove remarks

File: src/Components/Group/Selector/VersionView.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      const res = await itemRequest(path, name);
+      setItem(res);
+    } catch {
+      enqueueSnackbar(ERROR_CONTACTING_SERVER.message, {
+        variant: ERROR_CONTACTING_SERVER.variant,
+      });
+    }
+  };
+
+  // const selectItem = async function (version) {
```
Comment (asafMasa) [inline src/Components/Group/Selector/VersionView.js]: remove remarks

File: src/Components/Group/Selector/VersionView.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  return (
+    item && (
+      <TreeItem
+        key={path}
+        nodeId={item.latest.name + path}
+        label={item.latest.name}
+      >
+        {item.versions.map((version) => (
+          // <TreeItem
```
Comment (asafMasa) [inline src/Components/Group/Selector/VersionView.js]: remove remarks

File: src/Components/Group/Selector/VersionView.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  // const selectItem = async function (version) {
+  //   await getFromServer(path, version);
+  //   clickOnList({ path, item });
+  // };
+
+  return (
+    item && (
+      <TreeItem
+        key={path}
+        nodeId={item.latest.name + path}
```
Comment (asafMasa) [inline src/Components/Group/Selector/VersionView.js]: check if .id should be used

File: src/Components/Map/Map.js [hunk 11]
```diff
@@ -24,12 +28,17 @@ function App() {
     setVisible(!visible);
```
Comment (asafMasa) [inline src/Components/Map/Map.js]: check if INITIALIZE_MAP  is needed

File: src/Components/Map/Map.js [hunk 12]
```diff
@@ -24,12 +28,17 @@ function App() {
     setVisible(!visible);
   };
 
+  const fetchProject = async function (path, name) {
+    const res = await GetProject(path, name);
```
Comment (asafMasa) [inline src/Components/Map/Map.js]: camelCase

File: src/Components/Map/Map.js [hunk 13]
```diff
@@ -24,12 +28,17 @@ function App() {
     setVisible(!visible);
   };
 
+  const fetchProject = async function (path, name) {
+    const res = await GetProject(path, name);
+    dispatch({ type: INITIALIZE_STORE, payload: res.latest.resources });
```
Comment (asafMasa) [inline src/Components/Map/Map.js]: check name (INITIALIZE_STORE) validity

File: src/Config/urls.js [hunk 14]
```diff
@@ -1,14 +1,18 @@
-const baseUrl = "http://localhost:8000"; // 'https://gee.covid-97.com';
-const urlGetProjectByName = baseUrl + "/imagery/project/?name=";
-const urlThumbnail = baseUrl + "/imagery/resource/image/?";
-const urlGetProjects = baseUrl + "/imagery/projects/";
-const urlSearchProjects = baseUrl + "/imagery/project/search/";
-const urlSearchResources = baseUrl + "/imagery/resource/search/";
+const baseUrl = 'http://localhost:8000'; // 'https://gee.covid-97.com';
```
Comment (asafMasa) [inline src/Config/urls.js]: use config package

File: src/Requests/requests.js [hunk 15]
```diff
@@ -1,22 +1,51 @@
...
+  urlSearchProjects,
+  urlSearchResources,
+  urlGetProject,
+  urlGetResource,
+} from '../Config/urls';
 
-export async function GetProjects(path) {
-  const serachPath = path ? "?path=" + path : "";
-  const projects = await Axios.get(Config.urlGetProjects + serachPath);
+async function GetFromServer(url, path) {
```
Comment (asafMasa) [inline src/Requests/requests.js]: camelCase

File: src/Requests/requests.js [hunk 16]
```diff
@@ -1,22 +1,51 @@
...
+  const serachPath = path ? '?path=' + path : '';
+  const projects = await Axios.get(url + serachPath);
```
Comment (asafMasa) [inline src/Requests/requests.js]: try catch and write to log

File: src/Requests/requests.js [hunk 17]
```diff
@@ -1,22 +1,51 @@
...
+  return await GetFromServer(urlGetProjects, path);
+}
+
+export async function GetResources(path) {
+  return await GetFromServer(urlGetResources, path);
+}
+
+export async function Fetch(rootPath, path, name, version = 0) {
+  const query = version ? `?name=${name}&path=${path}&version=${version}` : `?name=${name}&path=${path}`;
+  const results = await Axios.get(rootPath + query);
```
Comment (asafMasa) [inline src/Requests/requests.js]: try catch and write to log
change to meaningful names

File: src/Store/Reducers/dataReducer.js [hunk 18]
```diff
@@ -165,31 +165,31 @@ export default function (state = defaultState, action) {
...
       });
     }
 
     case ADD_RESOURCE:
       return produce(state, (draft) => {
-        const resource = action.payload;
-        const groups = state.data["groups"];
-        const items = state.data.items;
+        const resource = action.payload.item;
+        const groups = draft.data["groups"];
```
Comment (asafMasa) [inline src/Store/Reducers/dataReducer.js]: '

File: src/Store/Reducers/dataReducer.js [hunk 19]
```diff
@@ -165,31 +165,31 @@ export default function (state = defaultState, action) {
...
         // TODO: should betolerance
-        draft.data.items[action.payload.name].mask.threshold =
+        draft.data.items[action.payload.id].mask.threshold =
           action.payload.tolerance;
-        draft.data.items[action.payload.name].mask.band = parseInt(
+        draft.data.items[action.payload.id].mask.band = parseInt(
           action.payload.band
         );
-        draft.data.items[action.payload.name].mask.white_fill = ~~action.payload
+        draft.data.items[action.payload.id].mask.white_fill = ~~action.payload
```
Comment (asafMasa) [inline src/Store/Reducers/dataReducer.js:178]: ~~?

File: src/Util/mapDataUtil.js [hunk 20]
```diff
@@ -8,17 +8,17 @@ export function addToGroups(groups, resource) {
       title: "Level " + resource.level,
       checked: true,
       level: resource.level,
-      itemsIds: [resource.name],
+      itemsIds: [resource.id],
     };
   } else {
-    groups[resource.level].itemIds.push(resource.name);
+    groups["level-" + resource.level].itemsIds.push(resource.id);
```
Comment (asafMasa) [inline src/Util/mapDataUtil.js]: '

File: src/Util/mapUtil.js [hunk 21]
```diff
@@ -76,28 +76,52 @@ export function addBaseLayer(map) {
 
   const osm = new OlLayerTile({
     source: new OlSourceOsm(),
-    name: "OSM",
+    name: 'OSM',
```
Comment (asafMasa) [inline src/Util/mapUtil.js]: const

File: src/Util/mapUtil.js [hunk 22]
```diff
@@ -76,28 +76,52 @@ export function addBaseLayer(map) {
...
   });
 
   mapLayers.push(layerGroupOsm);
 }
 
+export function addGroupToMap(map, newGroup) {
+  const layersArray = map.getLayers().getArray();
+  const index = layersArray.findIndex(
+    (group) =>
+      parseInt(group.get('name').replace('level-', '')) >=
```
Comment (asafMasa) [inline src/Util/mapUtil.js]: const groupPrefix

File: src/Util/treeUtil.js [hunk 23]
```diff
@@ -0,0 +1,3 @@
+export function createUniqueName(path, version) {
+    return version + path;
+}
```
Comment (asafMasa) [inline src/Util/treeUtil.js]: enter

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/Store/Reducers/dataReducer.js (hunk 19 above)
Comment (shimoncohen) [reply src/Store/Reducers/dataReducer.js:178]: Turns boolean to an integer value.

File: package.json [hunk 24]
```diff
@@ -13,6 +13,7 @@
     "@terrestris/react-geo": "^12.1.0",
     "antd": "^3.26.18",
     "axios": "^0.19.2",
+    "config": "^3.3.1",
```
Comment (asafMasa) [inline package.json]: revert

File: src/store/reducers/dataReducer.js [hunk 25]
```diff
@@ -177,15 +177,24 @@ export default function (state = defaultState, action) {
...
-          action.payload.sourceDate;
+        draft.data.items[action.payload.id].takenAt = action.payload.sourceDate;
       });
     }
 
     case ADD_RESOURCE:
       return produce(state, (draft) => {
         const resource = action.payload.item;
-        const groups = draft.data["groups"];
+        // const resource = {
```
Comment (asafMasa) [inline src/store/reducers/dataReducer.js]: remove remark

**Review by asafMasa — APPROVED** [review]

**Pull request merged**